source/material_model/rheology/visco_plastic.cc

```diff
@@ -479,6 +479,18 @@ namespace aspect
               composition_mask.set(i,false);
           }
 
+        // Mask fields that track the age and deposition depth of deposited sediments.
+        if (this->introspection().compositional_name_exists("sediment_age"))
+          {
+            const unsigned int sedi_age_position_tmp = this->introspection().compositional_index_for_name("sediment_age");
+            composition_mask.set(sedi_age_position_tmp,false);
+          }
+        if (this->introspection().compositional_name_exists("deposition_depth"))
+          {
+            const unsigned int depo_depth_position_tmp = this->introspection().compositional_index_for_name("deposition_depth");
+            composition_mask.set(depo_depth_position_tmp,false);
+          }
```

Comment:
This is unfortunate that you need to mask these field names in individual material models. Could this whole block be guarded somehow with an `if` that checks whether you use FastScape or not?